pkg/license/identifier.go

```diff
@@ -19,58 +19,48 @@ package license
 
 import (
 	"fmt"
-	"io/fs"
 	"path/filepath"
 	"regexp"
-	"strings"
 	"sync"
 
+	"github.com/google/licensecheck"
+
 	"github.com/apache/skywalking-eyes/assets"
 	"github.com/apache/skywalking-eyes/internal/logger"
 )
 
-var licenseTemplatesDir = "lcs-templates"
+const (
+	// coverageThreshold is the minimum percentage of the file that must contain license text.
+	// Reference: https://github.com/golang/pkgsite/blob/d43359e3a135fc391960db4f5800eb081d658412/internal/licenses/licenses.go#L48
+	coverageThreshold = 75
 
-var templatesDirs = []string{
-	licenseTemplatesDir,
-	// Some projects simply use the header text as their LICENSE content...
-	"header-templates",
-}
+	licenseTemplatesDir = "lcs-templates"
+)
 
-var dualLicensePatterns = []*regexp.Regexp{
-	regexp.MustCompile(`(?i)This project is covered by two different licenses: (?P<license>[^.]+)`),
-}
+var (
+	_scanner    *licensecheck.Scanner
+	scannerOnce sync.Once
 
-var normalizedTemplates = sync.Map{}
+	dualLicensePatterns = []*regexp.Regexp{
+		regexp.MustCompile(`(?i)This project is covered by two different licenses: (?P<license>[^.]+)`),
+	}
+)
 
-func init() {
-	wg := sync.WaitGroup{}
-	for _, dir := range templatesDirs {
-		files, err := assets.AssetDir(dir)
+// scanner returns a licensecheck.Scanner instance with its build-in licenses.
+// It will be initialized once.
+func scanner() *licensecheck.Scanner {
```

Comment:
can we initialize `_scanner` in `init` method? And I think Go doesn't require `_` prefix to indicate private variable?

Reply:
I prefer to lazy init the scanner:
1. It will cost more than one second to init the scanner
2. The scanner will only be used under `dep` command

Reply:
Yeah agree!